src/GafferImageUI/ImageView.cpp

```diff
@@ -275,8 +283,863 @@ GAFFER_NODE_DEFINE_TYPE( V2fContextVariable )
+			const V2f p = eventPosition( event );
+			if( onDeleteButton( p ) )
+			{
+				m_deleting = true;
+				return true;
+			}
+
+			return true;
+		}
+
+		IECore::RunTimeTypedPtr dragBegin( GafferUI::Gadget *gadget, const GafferUI::DragDropEvent &event )
+		{
+			m_dragStart = eventPosition( event );
+			m_dragStartPlugValue = m_plug->getValue();
+			return IECore::NullObject::defaultNullObject();
+		}
+
+		bool dragEnter( const GafferUI::Gadget *gadget, const GafferUI::DragDropEvent &event )
+		{
+			if( event.sourceGadget != this )
+			{
+				return false;
+			}
+
+			updateDragPoint( event );
+			return true;
+		}
+
+		bool dragMove( const GafferUI::DragDropEvent &event )
+		{
+			updateDragPoint( event );
+			return true;
+		}
+
+		bool dragEnd( const GafferUI::DragDropEvent &event )
+		{
+			updateDragPoint( event );
+			m_deleting = false;
+			return true;
+		}
+
+		void updateDragPoint( const GafferUI::DragDropEvent &event )
+		{
+			if( m_deleting )
+			{
+				return;
+			}
+			const V2f p = eventPosition( event );
+			V2i point = m_dragStartPlugValue;
+
+			const V2f offset = p - m_dragStart;
+			point += V2i( round( offset.x ), round( offset.y ) );
+
+			m_plug->setValue( point );
+		}
+
+		bool buttonRelease( const GafferUI::ButtonEvent &event )
+		{
+			const V2f p = eventPosition( event );
+			if( m_deleting && onDeleteButton( p ) )
+			{
+				m_deleteClickedSignal( m_plug.get() );
+			}
+			m_deleting = false;
+
+			return true;
+		}
+
+		void leave()
+		{
+			Pointer::setCurrent( "" );
+			m_hover = 0;
+			m_deleting = false;
+
+			Metadata::registerValue( m_plug.get(), g_hoveredKey, new IECore::BoolData( false ), false );
+		}
+
+		V2f viewportPlanarScale() const
+		{
+			// It's kinda silly to have to reverse engineer viewportGadget->m_planarScale because it isn't public
+			const ViewportGadget *viewportGadget = ancestor<ViewportGadget>();
+			return V2f(
+				viewportGadget->getCamera()->getAperture()[0] / viewportGadget->getViewport()[0],
+				viewportGadget->getCamera()->getAperture()[1] / viewportGadget->getViewport()[1]
+			);
+		}
+
+		bool onDeleteButton( const V2f &p ) const
+		{
+			// Any positions that are part of the gadget, but not part of an extended bound, are
+			// on the delete button
+			V2f point = V2f(m_plug->getValue()) + V2f( 0.5 );
+			const V2f planarScale = viewportPlanarScale();
+			const V2f threshold( planarScale * m_handleSize );
+			return
+				p.x > point.x + 0.5 * threshold.x - planarScale.x &&
+				p.y > point.y + 0.5 * threshold.y - planarScale.y;
+		}
+
+		V2f eventPosition( const ButtonEvent &event ) const
+		{
+			const ViewportGadget *viewportGadget = ancestor<ViewportGadget>();
+			const ImageGadget *imageGadget = static_cast<const ImageGadget *>( viewportGadget->getPrimaryChild() );
+			V2f pixel = imageGadget->pixelAt( event.line );
+			Context::Scope contextScope( imageGadget->getContext() );
+			pixel.x *= imageGadget->getImage()->format().getPixelAspect();
+			return pixel;
+		}
```

Comment:
Huh, yeah, I hadn't really looked at this at all, I just copied it from CropWindowTool.  It does seem fairly broken.  I guess it probably needs fixing in CropWindowTool too?

Is there any real utility to a Box2iGadget independent of being used on images?  Good UIs seem to need some awareness of their purpose rather than being 100% general anyway, and I can't think of two many places where we would need to manipulate a Box2i aside from the context of images?